src/sage/matrix/strassen.pyx

```diff
@@ -733,11 +731,9 @@ class int_range:
             sage: J - I
             [(6, 1), (20, 4)]
         """
-        all = self.to_list()
-        for i in right.to_list():
-            if i in all:
-                all.remove(i)
-        return int_range(all)
+        right_list = right.to_list()
```

Comment:
unless elements of `right_list` are not hashable, it would be better to make it a set to speed up tests in next line.